hazelcast/src/main/java/com/hazelcast/jet/impl/execution/OutboxImpl.java

```diff
@@ -256,7 +264,12 @@ final boolean offerToEdgesAndSnapshot(Object item) {
     }
 
     @Override
-    public long lastForwardedWm() {
-        return lastForwardedWm.get();
+    public long lastForwardedWm(byte key) {
+        Counter counter = lastForwardedWm.get(key);
```

Comment:
Should this metric-related method create new keys? Perhaps we can return MIN_VALUE.

If we decide to create new keys, we can use `computeIfAbsent`.